salt/utils/vault.py

```diff
@@ -126,24 +126,28 @@ def _get_vault_connection():
         return _get_token_and_url_from_master()
 
 
-def make_request(method, resource, profile=None, **args):
+def make_request(method, resource, token=None, vault_url=None, get_token_url=False, profile=None, **args):
     '''
     Make a request to Vault
     '''
     if profile is not None and profile.keys().remove('driver') is not None:
         # Deprecated code path
         return make_request_with_profile(method, resource, profile, **args)
 
-    connection = _get_vault_connection()
-    token, vault_url = connection['token'], connection['url']
-    if 'verify' not in args:
-        args['verify'] = connection['verify']
+    if not token or not vault_url:
+        connection = _get_vault_connection()
+        token, vault_url = connection['token'], connection['url']
+        if 'verify' not in args:
+            args['verify'] = connection['verify']
 
     url = "{0}/{1}".format(vault_url, resource)
     headers = {'X-Vault-Token': token, 'Content-Type': 'application/json'}
     response = requests.request(method, url, headers=headers, **args)
 
-    return response
+    if get_token_url:
+        return response, token, vault_url
```

Comment:
Do we need to handle this new return structure elsewhere in the code base?

Reply:
No, because it's not used anywhere. 
If this gets merged it would make sense to set get_token_url to True to get the token and then reuse it but not needed.